kubernetes/cache/istio.go

```diff
@@ -118,6 +125,23 @@ func (c *kialiCacheImpl) createIstioInformers(namespace string) istio.SharedInfo
 	return sharedInformers
 }
 
+func (c *kialiCacheImpl) createGatewayInformers(namespace string) gateway.SharedInformerFactory {
+	sharedInformers := gateway.NewSharedInformerFactory(c.gatewayApi, c.refreshDuration)
+	lister := c.getCacheLister(namespace)
+
+	if c.istioClient.IsGatewayAPI() {
```

Comment:
Something I wasn't expecting is that Kiali is automatically picking up the gateway API objects without a restart. It's also working fine without having the gateway API CRDs but I was expecting to have to restart the kiali pod before seeing the new Gateway objects.

Reply:
Restart of Kiali pod is required after adding (or removing) Gateway API CRDs only. Any further restart is not required while creating Gateway objects, as they are already registered in cache listeners.

Reply:
> Restart of Kiali pod is required after adding (or removing) Gateway API CRDs only. 

Yes! I was expecting to have to restart after adding the Gateway API CRDs but in fact Kiali picked them up automatically without needing to restart. I'm not sure if that's a good or bad thing to be honest. Something may be recreating the cache unexpectedly.

Reply:
Ah I figured out why these objects were showing up even with the Gateway APIs disabled. That info is coming from the registry.